### ISSUE TITLE
Instructions for Jellyseerr Integration?

### ISSUE THREAD

**Issue by FrostyDog9186** [hrfee/jfa-go#381]
I see the code, and the README talks about it, but there are no options in the UI, no directives in the INI or instructions for how to use the integration that I can find. How do I enable the integration?

**Comment by oxixes** [comment]
Seems like Jellyseerr is currently only available in the unstable channel.

**Comment by vonwanja** [comment]
Is there any information, when Jellyseerr implementation will merged into the stable release?

**Comment by devanteweary** [comment]
What integration are you talking about?  
My setup works fine but I wouldn't call it integration necessarily.

I send email invite.  
They sign up.  
This makes their Jellyfin account, which in turn they can use to log into Jellyseerr.  

They can also use either the Jellyseerr or Jellyfin forgot password links to reset their password.  

See this comment for a tip on how to hide the built-in Jellyfin password reset (which only works on the same network) and add a custom link instead: https://github.com/hrfee/jfa-go/issues/240#issuecomment-1779875680

**Comment by mkarimdev** [comment]
@devanteweary The integration we’re talking about isn’t just about creating a Jellyseerr account through Jellyfin. It’s about deeper customization and role-based control. In jfa-go, users can be grouped into "Profiles," which means we could actually assign Jellyseerr roles or privileges based on those profiles. Things like limiting the number of requests, allowing only certain qualities, etc.

Another issue is that Jellyseerr only pulls from Jellyfin, which doesn’t store emails—just usernames. Since jfa-go has emails, it could fix that by adding them directly to the Jellyseerr database. This would make account management way smoother.

An integration like this could open up a lot of useful options. Of course, dev has his own priorities, so hopefully, the original dev or some-hero else who knows Go can give this a shot. I wish I could help, but I’m a Go noob :")

**Comment by devanteweary** [comment]
Oh OK gotcha.

> Another issue is that Jellyseerr only pulls from Jellyfin, which doesn’t store emails—just usernames. Since jfa-go has emails, it could fix that by adding them directly to the Jellyseerr database. This would make account management way smoother.

Yeah this _would_ be nice.  
Right now the way I have it set up is Graylog checks the logs and whenever someone creates their jfa-go / Jellyfin account, sends me an alert of things I need to do such as add their email address in Jellyseerr ha

So hopefully!

**Comment by hrfee** [comment]
@mkarimdev @devanteweary i honestly don't remember but i think this is one of the few things that the jfa-go integration does (importing emails and other contact methods from jfa-go into jellyseerr), there's a daemon portion of the integration as i remember which i assume serves this purpose.

i'll write a jellyseerr wiki page too, i recall writing one in the past but maybe i'm misremembering.

**Comment by mkarimdev** [comment]
@hrfee Thanks for the clarification! If the daemon part already handles importing emails and contact methods from jfa-go into Jellyseerr, that’s great. But from what I can see, the current Docker release of jfa-go doesn’t seem to include the Jellyseerr integration—only Ombi.

It’s a bit confusing since the Jellyseerr integration was committed over 7 months ago, but it’s not available in the latest build. Do you know if it’s only in the unstable channel or if it just hasn’t been included in a release yet?

A Jellyseerr wiki page would be really helpful—especially if it covers what’s actually implemented and how to enable it. Thanks again for all your work on this!

**Comment by hrfee** [comment]
@mkarimdev yeah, I haven't tagged a stable release in a long time since I haven't touched the project in a while, and to be honest I'm unlikely to do one for another while so I recommend you switch to unstable and keep a keen eye on what each update/commit is, and turn on regular DB backups. I've just pushed a little paved to the wiki describing what the Jellyseerr integration does too.

**Comment by bin101** [comment]
@hrfee tried to use the jellyseerr integration with the unstable docker image but sadly it seems to not work. Debug logs show this:
```
[DEBUG] 13:15:44 jellyseerr-d.go:14: Failed to get or trigger import for Jellyseerr user "b9797e0d87ff4223a6cb6d8f2eff5946": json: cannot unmarshal object into Go value of type []jellyseerr.User
[DEBUG] 13:15:45 jellyseerr-d.go:14: Failed to get or trigger import for Jellyseerr user "1c8954ea27f646fc94e8352f52badd3d": json: cannot unmarshal object into Go value of type []jellyseerr.User
[DEBUG] 13:15:45 jellyseerr-d.go:14: Failed to get or trigger import for Jellyseerr user "dd5f4db6f5524eff99e995fe719e6a16": json: cannot unmarshal object into Go value of type []jellyseerr.User
[DEBUG] 13:15:45 jellyseerr-d.go:14: Failed to get or trigger import for Jellyseerr user "b85eb1ac5d2443189fc12a2b34d8cc2a": json: cannot unmarshal object into Go value of type []jellyseerr.User
```
Those are 4 new user, but the old ones also doesn't have their mail addresses sync.

**Comment by bin101** [comment]
It is definitely the correct api key and the debug output above is telling about deserialization issues. My jellyseer instance is currently at version 2.5.2, maybe something changed inside the data structure?

**Comment by hrfee** [comment]
I'm not sure what else it could be, would you be willing to do a man-in-the-middle sort of thing between jfa-go and Jellyseerr to see what's actually happening? I wrote some instructions [on the jfa-go-wiki's developer section](https://wiki.jfa-go.com/docs/dev/#debugging-api-clients-mb-jellyfinemby-ombi-jellyseerr), You essentially point jfa-go at mitmproxy, and point mitmproxy at jellyseerr. It's UI will then let you see each request made, and ideally see what data is returned by Jellyseerr that jfa-go can't parse.

**Comment by bin101** [comment]
@hrfee thanks for the proxy hint, it was totally my fault. Had CSRF Protection enabled in jellyseer which made all api calls read-only. But you could add a warning that this needs to be disabled in jellyseer if someone wants to use the sync 😉.
